python/friesian/democode/train_2tower.py

```diff
@@ -0,0 +1,175 @@
+    return estimator
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(description='Two Tower Training/Inference')
+    parser.add_argument('--cluster_mode', type=str, default="local",
+                        help='The cluster mode, such as local, yarn, standalone or spark-submit.')
+    parser.add_argument('--master', type=str, default=None,
+                        help='The master url, only used when cluster mode is standalone.')
+    parser.add_argument('--executor_cores', type=int, default=8,
+                        help='The executor core number.')
+    parser.add_argument('--executor_memory', type=str, default="160g",
+                        help='The executor memory.')
+    parser.add_argument('--num_executor', type=int, default=8,
+                        help='The number of executor.')
+    parser.add_argument('--driver_cores', type=int, default=4,
+                        help='The driver core number.')
+    parser.add_argument('--driver_memory', type=str, default="36g",
+                        help='The driver memory.')
+    parser.add_argument('--lr', default=0.001, type=float, help='learning rate')
+    parser.add_argument('--epochs', default=1, type=int, help='train epoch')
+    parser.add_argument('--batch_size', default=8000, type=int, help='batch size')
+    parser.add_argument('--model_dir', default='snapshot', type=str,
+                        help='snapshot directory name (default: snapshot)')
+    parser.add_argument('--data_dir', type=str, default="./movielens/processed", help='data directory')
+
+    args = parser.parse_args()
+
+    if args.cluster_mode == "local":
+        sc = init_orca_context("local", init_ray_on_spark=True)
+    elif args.cluster_mode == "standalone":
+        sc = init_orca_context("standalone", master=args.master,
+                               cores=args.executor_cores, num_nodes=args.num_executor,
+                               memory=args.executor_memory,
+                               driver_cores=args.driver_cores, driver_memory=args.driver_memory,
+                               conf=spark_conf,
+                               init_ray_on_spark=True)
+    elif args.cluster_mode == "yarn":
+        sc = init_orca_context("yarn-client", cores=args.executor_cores,
+                               num_nodes=args.num_executor, memory=args.executor_memory,
+                               driver_cores=args.driver_cores, driver_memory=args.driver_memory,
+                               conf=spark_conf, extra_python_lib="../example/two_tower/model.py",
+                               object_store_memory="8g",
+                               init_ray_on_spark=True)
+    elif args.cluster_mode == "spark-submit":
+        sc = init_orca_context("spark-submit")
+    else:
+        invalidInputError(False,
+                          "cluster_mode should be one of 'local', 'yarn', 'standalone' and"
+                          " 'spark-submit', but got " + args.cluster_mode)
```

Comment:
this is too complex; there is no need to demonstrate all the possible cluster modes in this example.